homeassistant/components/modbus/climate.py

```diff
@@ -106,17 +154,32 @@ def supported_features(self):
     @property
     def hvac_mode(self):
         """Return the current HVAC mode."""
-        return HVAC_MODE_AUTO
+        return self._hvac_mode
 
     @property
     def hvac_modes(self):
         """Return the possible HVAC modes."""
-        return [HVAC_MODE_AUTO]
+        return self._hvac_mode_supported
 
     async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         """Set new target hvac mode."""
-        # Home Assistant expects this method.
-        # We'll keep it here to avoid getting exceptions.
+        if hvac_mode in self._hvac_mode_dict and self._hvac_mode_register != 0:
+            mode_value = self._hvac_mode_dict[hvac_mode]
```

Comment:
you cannot assume our internal hvac_mode is what the modbus device needs.

Reply:
I'm not sure I understand - this function is what home-assistant calls from the UI (or elsewhere) to update the device's HVAC mode. If we have a register defined in the config file, and register values that have been mapped to home assistant's `HVAC_MODE`s, then we should be able to use this functionality for modbus devices. 

If I have a control loop (thermal or otherwise) on a modbus device, I should be able to set it to something like 'cool', or 'heat', or 'off' from the UI. This wouldn't be any different than a home thermostat being changed from 'heat' to 'off'.